### PR TITLE
Offboarding users when they have an out of state W2

### PR DESCRIPTION
**Pull request by tofarr**
When a user has a W2 from outside the current state in their direct file data, they get the following page:
image

To get tests working again after adding this additional constraint, I had to add a new test dataset that mimics an existing one, but has the state as AZ instead of NY

### CODE REVIEW

File: app/models/state_file_az_intake.rb [hunk 1]
```diff
@@ -135,7 +135,13 @@ def ask_spouse_dob?
   end
 
   def disqualifying_df_data_reason
-    :married_filing_separately if direct_file_data.filing_status == 3
+    return :married_filing_separately if direct_file_data.filing_status == 3
+
+    w2_states = direct_file_data.parsed_xml.css('W2StateLocalTaxGrp W2StateTaxGrp StateAbbreviationCd')
+    :has_out_of_state_w2 if w2_states.any? do |state|
+      (state.text || '').upcase != 'AZ'
```
Comment (rickreyhsig) [inline app/models/state_file_az_intake.rb:142]:
```suggestion
      (state.text || '').upcase != state_code.upcase
```

File: app/models/state_file_ny_intake.rb [hunk 2]
```diff
@@ -196,6 +196,11 @@ def ach_debit_transaction?
   IRC_125_CODES = ['IRC125S', 'IRS125']
   YONKERS_CODES = ['YK', 'YON', 'YNK', 'CITYOFYK', 'CTYOFYKR', 'CITYOF YK', 'CTY OF YK']
   def disqualifying_df_data_reason
+    w2_states = direct_file_data.parsed_xml.css('W2StateLocalTaxGrp W2StateTaxGrp StateAbbreviationCd')
+    return :has_out_of_state_w2 if w2_states.any? do |state|
+      (state.text || '').upcase != 'NY'
```
Comment (rickreyhsig) [inline app/models/state_file_ny_intake.rb:201]:
```suggestion
      (state.text || '').upcase != state_code.upcase
```

File: spec/features/state_file/complete_intake_spec.rb [hunk 3]
```diff
@@ -165,7 +165,7 @@
       expect(page).to have_text I18n.t('state_file.questions.terms_and_conditions.edit.title')
       click_on I18n.t("state_file.questions.terms_and_conditions.edit.accept")
 
-      step_through_df_data_transfer
+      step_through_df_data_transfer("Transfer AZ Old sample")
```
Comment (rickreyhsig) [inline spec/features/state_file/complete_intake_spec.rb:168]: Are you using the string for label purposes?

File: app/models/state_file_az_intake.rb [hunk 4]
```diff
@@ -135,7 +135,13 @@ def ask_spouse_dob?
...
+    end
+
```
Comment (rickreyhsig) [inline app/models/state_file_az_intake.rb:144]: Nit: extra line

**Review by rickreyhsig — APPROVED** [review]
LGTM! Left a few suggestions.

File: spec/features/state_file/complete_intake_spec.rb (hunk 3 above)
Comment (tofarr) [reply spec/features/state_file/complete_intake_spec.rb:168]: The string is actually used to select the button to click in the test. This was hardcoded to 
```click_on "Transfer my 2023 federal tax return to FileYourStateTaxes"```
 so I added an optional parameter to allow clicking on one of the other elements instead. (See state_file_intake_helper.rb Line 70

File: app/models/state_file_az_intake.rb (hunk 4 above)
Comment (tofarr) [reply app/models/state_file_az_intake.rb]: Removed

**Pull request merged**